File: TypeScript/single_number/136.ts
function singleNumber(nums: number[]): number {
  const map = new Map<number, number>();

  nums.forEach((num) => {
    const v = map.get(num);
    if (v == undefined) {
      map.set(num, 1);
    } else {
      map.set(num, v + 1);
    }
  });

  for (let i = 0; i < nums.length; i++) {
    const v = map.get(nums[i]);
    if (v === 1) {
      return nums[i];
    }
  }

  return -1;
}

export { singleNumber };

File: Golang/single_number/136.go
package singlenumber

func singleNumber(nums []int) int {
	hash_map := make(map[int]int)
	for i := 0; i < len(nums); i++ {
		if value, ok := hash_map[nums[i]]; ok {
			hash_map[nums[i]] = value + 1
		} else {
			hash_map[nums[i]] = 1
		}
	}

	result := 0
	for i := 0; i < len(nums); i++ {
		if hash_map[nums[i]] == 1 {
			result = nums[i]
		}
	}

	return result
}
